### DISCUSSION TITLE
Model selection.

### DISCUSSION THREAD

**Discussion by jmchen5821** [threeML/threeML#480] [Q&A]
Hi! I am using Bayesian MCMC to analyze spectra. Which statistical measurement values (AIC, BIC, DIC) should I use to select the model?

**Comment by grburgess** [comment]
In general, even though we report all of these values... one should steer away from the BIC. The AIC should not be used with bayesian fits. 

DIC is a predictive measure. It can only tell you that a model relatively better predicts the data than another one... but not how much better. It also has caveats as it is slow to converge. 

In general, posterior predictive checks are are good way to assess the quality of fits. There is a package that does this for some data with 3ML: https://twopc.readthedocs.io/en/latest/

Model selection is a tricky business within the world of statistics and literature is a good place to start. 

I recommend this article: https://projecteuclid.org/journals/statistics-surveys/volume-6/issue-none/A-survey-of-Bayesian-predictive-methods-for-model-assessment-selection/10.1214/12-SS102.full

**Comment by jmchen5821** [comment]
Thank you very much for your reply. Sometimes DIC and PDIC are negative, how to compare with other models?

**Comment by grburgess** [comment]
if PDIC (the effective number of parameters) is negative, this is a sign that the median of the posterior is far from the mode. Thus, it will be difficult to do model comparison.

**Comment by jmchen5821** [comment]
Thank you very much, your answer is of great help to me !

**Comment by kchoX14** [comment]
> In general, even though we report all of these values... one should steer away from the BIC. The AIC should not be used with bayesian fits.
> 
> DIC is a predictive measure. It can only tell you that a model relatively better predicts the data than another one... but not how much better. It also has caveats as it is slow to converge.
> 
> In general, posterior predictive checks are are good way to assess the quality of fits. There is a package that does this for some data with 3ML: https://twopc.readthedocs.io/en/latest/
> 
> Model selection is a tricky business within the world of statistics and literature is a good place to start.
> 
> I recommend this article: https://projecteuclid.org/journals/statistics-surveys/volume-6/issue-none/A-survey-of-Bayesian-predictive-methods-for-model-assessment-selection/10.1214/12-SS102.full

It seems like PPC is quite similar to parametric bootstrapping with MCMC simulations (using the goodness of fit plugin of threeML), except that GoF bootstrap takes the current model while PPCs sample directly over the posterior. Am I right is thinking so? Moreover, the PPC check with "twopc" seems to be more of a visual check (like cumulative QQ plots of observed vs model predicted counts in BXA). It can get ugly if the fits involve multiple instruments and/or spectra. It is also not clear why testing twice (once with source+background and once with the background-subtracted spectrum) is necessary for a visual model selection. Could it be used if backgrounds are modeled separately instead of assuming a joint fit?

Given it is a visual alternative to AIC, BIC or DIC, it complicates the model comparison perspective since numbers aren't involved. DIC is only applicable if the posterior is multi-normal for sure (like one uses with MCMC over XSPEC or emcee on python), and AIC has a frequentist base. [https://stats.stackexchange.com/questions/115157/what-are-posterior-predictive-checks-and-what-makes-them-useful](https://stats.stackexchange.com/questions/115157/what-are-posterior-predictive-checks-and-what-makes-them-useful) suggests PPCs can be use to check if the model fits the data like a Monte Carlo GoF bootstrapping (as aforementioned).

In light of all this, how would a crowded (multiple spectra-instruments), visual PPC be indicative of a better fit and serve as model comparison (other than the evidence)?

**Comment by grburgess** [comment]
Indeed, PPCs are not really a selection tool, but a model fitness tool. There are ways to compress PPCs into numerical checks by computing Bayesian tail probabilities, for example. However, these are not well understood as their distribution is not guaranteed to be uniform ( https://projecteuclid.org/journals/electronic-journal-of-statistics/volume-7/issue-none/Two-simple-examples-for-understanding-posterior-p-values-whose-distributions/10.1214/13-EJS854.full )

Model selection is very tricky. One thing that is known for sure is that marginal likelihoods are not a quick alternative, but there are appropriate ways to use them (https://arxiv.org/abs/2103.08744)


PPCs are related to bootstrap goodness-of-fit checks... but derived from a mathematically generic and robust approach. Two good resources for this are: https://arxiv.org/abs/1709.01449 and https://betanalpha.github.io/assets/case_studies/principled_bayesian_workflow.html

**Comment by kchoX14** [comment]
> Indeed, PPCs are not really a selection tool, but a model fitness tool. There are ways to compress PPCs into numerical checks by computing Bayesian tail probabilities, for example. However, these are not well understood as their distribution is not guaranteed to be uniform ( https://projecteuclid.org/journals/electronic-journal-of-statistics/volume-7/issue-none/Two-simple-examples-for-understanding-posterior-p-values-whose-distributions/10.1214/13-EJS854.full )
> 
> Model selection is very tricky. One thing that is known for sure is that marginal likelihoods are not a quick alternative, but there are appropriate ways to use them (https://arxiv.org/abs/2103.08744)
> 
> 
> PPCs are related to bootstrap goodness-of-fit checks... but derived from a mathematically generic and robust approach. Two good resources for this are: https://arxiv.org/abs/1709.01449 and https://betanalpha.github.io/assets/case_studies/principled_bayesian_workflow.html
> 

I understand PPCs are a crude visualization tool like BXA's QQ plots. But why is there a need to check once with no background involved and again with the background-subtracted spectrum? How much of a deviation of the PPC fit against the posterior data is significant enough to compare two models (nested and non-nested alike) and rule out one over the other? There should be some simple factor or normalization, right? Bayes Factor (taking the ratio of evidences Z) is good to compare on a crude basis for model comparison of quality. But it'd require additional support from the residual plots and the PPC plots (if not AIC).

**Comment by grburgess** [comment]
@kchoX14 I'm not sure what you mean by a two step process.

**Comment by kchoX14** [comment]
> @kchoX14 I'm not sure what you mean by a two step process. 

I mean, in the **twopc** page you have, you show that the PPC tests have been done twice:  once over the background-subtracted spectrum, and again over just the source+background spectrum. That is what I meant by a two-step process.

**Comment by grburgess** [comment]
These are just options. For very dominant backgrounds, the uncertainties in the background can overwhelm the uncertainties in the source. Thus, in those cases, deviations from the model are easier to see via subtraction.

**Comment by kchoX14** [comment]
So, my original query remains:  how much of a deviation seen in the PPC plots are significant to visually test the goodness of the model versus another?
